src/streamsync/core.py

```diff
@@ -215,19 +217,7 @@ def __setitem__(self, key, raw_value) -> None:
             raise ValueError(
                 f"State keys must be strings. Received {str(key)} ({type(key)}).")
 
-        # Items that are dictionaries are converted to StateProxy instances
-
-        if isinstance(raw_value, dict):
-            value = StateProxy(raw_value)
-        elif isinstance(raw_value, StateProxy):
-            # Children StateProxies need to be reinitialised
-            # during an assignment to parent StateProxy
-            # to ensure proper mutation tracking
-            value = StateProxy(raw_value.state)
-        else:
-            value = raw_value
-
-        self.state[key] = value
+        self.state[key] = raw_value
```

Comment:
I have to move StateProxy creation in the State because, it's this object that know the schema. You should take a look on the method `State._set_state_item`.